internal/kuberesource/wrappers.go

```diff
@@ -31,15 +33,29 @@ func (t *PodTemplate) GetObjectMeta() *applymetav1.ObjectMetaApplyConfiguration
 	if t.PodTemplateSpecApplyConfiguration != nil && t.ObjectMetaApplyConfiguration != nil {
 		return t.ObjectMetaApplyConfiguration
 	}
-	return &applymetav1.ObjectMetaApplyConfiguration{}
+	return nil
+}
+
+// SetObjectMeta sets the ObjectMeta of the Pod template.
+func (t *PodTemplate) SetObjectMeta(meta *applymetav1.ObjectMetaApplyConfiguration) {
+	if t.PodTemplateSpecApplyConfiguration != nil {
+		t.ObjectMetaApplyConfiguration = meta
+	}
```

Comment:
It's a bit unexpected that `SetObjectMeta` does nothing if the wrapped template is nil. Maybe we should initialize it? Applies to other methods below.